doc/contribute/bin_blobs.rst

```diff
@@ -0,0 +1,201 @@
+
+Note that just because a blob has an acceptable type does not imply that it will
+be unconditionally accepted by the project; any blob may be rejected for other
+reasons on a case by case basis.
+In case of disagreement, the TSC is the arbiter of whether a particular blob
+fits in one of the above types.
+
+Hardware agnostic features provided via a proprietary library are not
+acceptable. For example, a proprietary and hardware agnostic TCP/IP stack
+distributed as a static archive would be rejected.
+
+Precompiled library-specific requirements
+=========================================
+
+This section contains additional requirements specific to precompiled library
+blobs.
+
+Any person who wishes to submit a precompiled library must represent that it
+meets these requirements. The project may remove a blob from the upstream
+distribution if it is discovered that the blob fails to meet these requirements
+later on.
+
+Interface header files
+----------------------
+
+The precompiled library must be accompanied by one or more header files,
+distributed under an OSI approved license, that define the interface to the
+library.
+
+Allowed dependencies
+--------------------
+
+This section defines requirements related to external symbols that a library
+blob requires the build system to provide.
+
+* The blob must not depend on Zephyr APIs directly. This is required for loose
+  coupling and maintainability, since Zephyr APIs may change and such blobs
+  cannot be modified by all project maintainers
+* Instead, an implementation of a porting layer defining all such symbols can be
+  provided alongside the library if required. The implementation must be in
+  source code form, released under an OSI approved license
+
+Toolchain requirements
+----------------------
+
+Library blobs must be in a data format which is compatible with a toolchain
+supported by the Zephyr Project. This is required for maintainability and
+usability. Use of such libraries may require special compiler and/or linker
+flags, however. For example, a porting layer may require special flags, or a
+static archive may require use of specific linker flags.
+
+Limited scope
+-------------
+
+Allowing arbitrary library blobs would carry a risk of degrading the degree to
+which the upstream Zephyr software distribution is open source. As an extreme
+example, a target with a zephyr kernel clock driver that is just a porting layer
+around a library blob would not be bootable with open source software.
+
+To mitigate this risk, the scope of upstream library blobs will be limited. The
+project maintainers must define an open source test suite that an upstream
+target must be able to pass using only open source software included in the
+mainline distribution and its modules. The scope of this test suite may grow
+over time, but the goal is to specify tests for a minimal feature set which must
+be supported via open source software for any target with upstream Zephyr
+support.
```

Comment:
In this Pull Request we should decide on this particular point, or at least on an initial version of it.

I suggest we require all platforms to be able to run the `tests/kernel` suite without the use of binary blobs.

Reply:
That sounds like a good starting point. Perhaps we can have a special "opensource" `tag` in the testcase.yml files to allow this to be easily ran using twister.

Reply:
I will start with:

- `samples/philosophers`
- `tests/kernel`